src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -175,7 +181,8 @@ public SlaveTemplate(String ami, String zone, SpotConfiguration spotConfig, Stri
             String instanceCapStr, String iamInstanceProfile, boolean deleteRootOnTermination,
             boolean useEphemeralDevices, boolean useDedicatedTenancy, String launchTimeoutStr, boolean associatePublicIp,
             String customDeviceMapping, boolean connectBySSHProcess, boolean monitoring,
-            boolean t2Unlimited, ConnectionStrategy connectionStrategy, int maxTotalUses) {
+            boolean t2Unlimited, BurstableUnlimitedMode burstableUnlimitedMode, ConnectionStrategy connectionStrategy,
```

Comment:
I wonder whether `t2Unlimited` should maybe be removed completely from this constructor and be made into a (deprecated) `DataBoundSetter` instead (to make it clear that `t2Unlimited` is deprecated; its presence in the `DataBoundConstructor` signature might be confusing).

Reply:
I just realized that this would break the migration of `t2Unlimited`, unless we move that from `readResolve()` back to e. g. `getBurstableUnlimitedMode()`... So let's leave things like they are for now.

Reply:
You can do it this way, it removes the Deprecated field from the ctor to make it less confusing.
https://github.com/jenkinsci/ec2-plugin/blob/597af44906321da78a92fd355355b77f5751a679/src/main/java/hudson/plugins/ec2/SlaveTemplate.java#L423

Reply:
Ah, right. I will look into that.